### DISCUSSION TITLE
XenAPI.XenAPI.Failure: ['SESSION_AUTHENTICATION_FAILED', 'root', 'Authentication failure']

### DISCUSSION THREAD

**Discussion by Winiron-15** [MikeDombo/xen-exporter#11] [Q&A]
Hello,

When I install the Docker container and navigate to its logs, it shows me that it failed to authenticate (see the screenshot below). Could I get some assistance with this ?

![Capture d'écran 2024-04-11 183019](https://github.com/MikeDombo/xen-exporter/assets/128186911/606b9b0d-841f-4dfd-be54-ad96bf633d3c)

**Comment by MikeDombo** [comment]
Follow the instructions to specify your correct username and password: https://github.com/MikeDombo/xen-exporter?tab=readme-ov-file#usage.

**Reply by Winiron-15** [reply]
The issue is that I've already specified the correct username and password.

**Reply by MikeDombo** [reply]
That is not what the error says, the error is coming directly from Xen and this script cannot do anything about that.

**Reply by Winiron-15** [reply]
I don't know, but I've precisely entered the correct username and password in my Docker-compose file.
![2024-04-11 18_47_47-Portainer _ srv-docker-PROD - Opera](https://github.com/MikeDombo/xen-exporter/assets/128186911/91df56de-b9ff-4762-9c31-8858e03b3f7e)

**Reply by Winiron-15** [reply]
I connect via SSH to my XCP-ng server using the same credentials.

**Reply by MikeDombo** [reply]
See if you can print out the password here and ensure it matches your expectation: https://github.com/MikeDombo/xen-exporter/blob/master/xen-exporter.py#L103

**Reply by MikeDombo** [reply]
Also try using the map format for environment instead of list: https://www.reddit.com/r/docker/comments/sva0yf/comment/hxg30ur/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

**Reply by Winiron-15** [reply]
I tried changing the format of the environment variables, but it didn't make any difference, and I'm not sure how to modify and execute the code. I'm fairly new to computing.

**Reply by MikeDombo** [reply]
How about this. Can you try changing your password to just be normal alphabetic characters? This will tell us more about where the problem is. You can change your password back again after seeing if we can connect using a simple password.

**Reply by Winiron-15** [reply]
I removed the special characters, and it worked. Thank you for the time you took to help me.
